src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -186,7 +186,7 @@ protected override Task<Stream> CreateContentReadStreamAsync()
 
         protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
         {
-            return _messageEncoder.WriteMessageAsync(_message, _stream);
+            return _messageEncoder.WriteMessageAsync(_message, stream);
```

Comment:
The code being replaced looks odd.  It was serializing the _stream instance member, not the passed in stream.  If it turns out they are the same stream, I recommend adding a Contract.Assert(Object.ReferenceEquals(stream, _stream)).